src/client/game.cpp

```diff
@@ -1220,6 +1220,7 @@ bool Game::startup(bool *kill,
 
 	if (!createClient(start_data))
 		return false;
+	input->registerLocalPlayer(client->getEnv().getLocalPlayer());
```

Comment:
The inputHandler abstraction wasn't made for what we're doing with it. The Remote input handler needs a reference to the layer player so that the handler can access and send back the reward, which is passed via HUD.